File: 1430-find-the-k-beauty-of-a-number/find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        
        c = 0
        for i in range(len(str(num))-k+1):
            subStr = str(num)[i:i+k]
            print(subStr)

            # if len(subStr) <k:
            #     break
            if int(subStr) == 0:
                continue
            if num%int(subStr) == 0:
                c+=1
        print(c)
        return c

File: 1430-find-the-k-beauty-of-a-number/find-the-k-beauty-of-a-number.java
class Solution {
    public int divisorSubstrings(int num, int k) {

        int c=0;
        String numString = ""+num;

        StringBuilder window = new StringBuilder();

        String kNumStr = numString.substring(0,k);
        window.append(kNumStr);
        int knum = Integer.parseInt(kNumStr);

        if(num%knum == 0){
            c++;
        }

        for(int j=k;j<numString.length();j++){
            window.deleteCharAt(0);
            window.append(numString.charAt(j));
            kNumStr = window.toString();
            knum = Integer.parseInt(kNumStr);
            if(knum!=0 && num%knum == 0){
                c++;
            }


        }






        return c;

    }
}